core/arch/arm/plat-imx/imx.h

```diff
@@ -42,12 +42,5 @@ void imx_gpcv2_set_core1_pdn_by_software(void);
 void imx_gpcv2_set_core1_pup_by_software(void);
 void imx_gpcv2_set_core_pgc(bool enable, uint32_t offset);
 
-#ifdef CFG_TZC380
-void imx_configure_tzasc(void);
-#else
-static inline void imx_configure_tzasc(void)
-{
-}
-#endif /* CFG_TZC380 */
```

Comment:
There will be a problem with uboot-imx by doing this. The TZASC will be configured after the memory mapping of the DTB (non-secure).

But this is upstream, so you will have no problem with uboot upstream :man_shrugging:
I just want to remind people that this might not work with uboot OEM.

So, you changed your mind about the tzasc initialization ?
https://github.com/OP-TEE/optee_os/pull/3017#discussion_r285082801

Reply:
> There will be a problem with uboot-imx by doing this. The TZASC will be configured after the memory mapping of the DTB (non-secure).

How does uboot-imx configure the regions? For barebox I currently configure region 0 to be accessible by both normal and secure world. We should establish a convention here, document it and push for implementation in all bootloaders.

> But this is upstream, so you will have no problem with uboot upstream man_shrugging
> I just want to remind people that this might not work with uboot OEM.

If possible and not to maintanance intensive, I'd like to keep compatiblity, see my reasoning above.
> 
> So, you changed your mind about the tzasc initialization ?
> [#3017 (comment)](https://github.com/OP-TEE/optee_os/pull/3017#discussion_r285082801)

Yes, due to running into the same issue with the external DTB, I have since added the small line to Barebox to configure region 0 to be accessible from the normal and secure world. I also wanted the dump functionality to work, but since `plat_cpu_reset_late` runs before the console, moving it to `driver_init` seemed like a good viable option.

Reply:
> > There will be a problem with uboot-imx by doing this. The TZASC will be configured after the memory mapping of the DTB (non-secure).
> 
> How does uboot-imx configure the regions? For barebox I currently configure region 0 to be accessible by both normal and secure world. We should establish a convention here, document it and push for implementation in all bootloaders.

uboot-imx just enables the TZASC and by default, region 0 maps all the DDR in secure. The DDR remains in secure until OPTEE configures tzc regions.

Reply:
Would it be possible to modify downstream U-Boot to also set the permissions on region 0 to secure and non-secure? Otherwise we can't really make use of the tzc_dump functionality.

Reply:
That's something we might do indeed. We don't mind changing the region 0 from secure to non-secure in uboot-imx.
This way, optee-os will be compatible with uboot, uboot-imx and Barebox